core/src/mindustry/world/blocks/production/GenericCrafter.java

```diff
@@ -15,7 +15,9 @@
 import mindustry.world.meta.*;
 
 public class GenericCrafter extends Block{
+    /** @deprecated use outputItems instead, outputItem would likely be removed eventually **/
     public @Nullable ItemStack outputItem;
```

Comment:
`outputItem` is too convenient and too widely used to deprecated. Instead, use **only** `outputItems` in the update code, and make `outputItems` a single-item array if it is null and `outputItems != null` in `init()`.

Reply:
Also worth noting that this implementation doesn't handle `shouldConsume` properly.

Reply:
Isn't `outputsItems` also not handled correctly?

Reply:
yes